synapse/handlers/device.py

```diff
@@ -1175,3 +1203,33 @@ async def process_cross_signing_key_update(
             device_ids.append(verify_key.version)
 
         return device_ids
+
+    async def handle_room_un_partial_stated(self, room_id: str) -> None:
+        """Handles sending appropriate device list updates in a room that has
+        gone from partial to full state.
+        """
+
+        pending_updates = (
+            await self.store.get_pending_remote_device_list_updates_for_room(room_id)
+        )
+
+        for user_id, device_id in pending_updates:
+            logger.info(
+                "Got pending device list update in room %s: %s / %s",
+                room_id,
+                user_id,
+                device_id,
+            )
+            position = await self.store.add_device_change_to_streams(
+                user_id,
+                [device_id],
+                room_ids=[room_id],
+            )
+
+            if not position:
+                # This should only happen if there are no updates, so we bail.
+                continue
```

Comment:
This is unreachable, isn't it?

Reply:
Err, yeah actually. Should we do something else here?

Reply:
Can we adjust the comment to explain that it's unreachable?